### PR TITLE
NAS-117317 / 22.12 / Smaller fixes for Storage pages

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Tests:
1. In Datasets in Space Management there is a line Quotas set for 1 users. Use ICU message format to show different message for 1 vs many items. I.e. set for 1 user and set for 2 users. Same for groups.

2. When Datasets tree is expanded and you scroll down last row ends up being too close to bottom edge of the window. Check if adding margin in the bottom of datasets tree makes it look better.
Do the same in Devices.

3. When Zvol is selected, update title of Dataset Details card to Zvol Details.

4. Ask @Damian Szidiropulosz for an icon for Datasets menu item in main menu and update the icon.

5. In Datasets it seems that only first level is sorted alphabetically. Make sure children inside a datasets are also sorted alphabetically.
Do the same in Devices.

### CODE REVIEW

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html [hunk 1]
```diff
@@ -64,9 +64,16 @@ <h3 mat-card-title>
       <div class="details-item">
         <span class="label">{{ 'User Quotas' | translate }}:</span>
         <ng-container *ngIf="checkQuotas; else noneTemplate">
-          <span class="value" *ngIf="!isLoadingQuotas; else valuePlaceholder">
-            {{ 'Quotas set for {n} users' | translate: { n: userQuotas } }}
-          </span>
+          <div *ngIf="!isLoadingQuotas; else valuePlaceholder">
+            <span class="value" *ngIf="userQuotas === 1; else manyQuotas">
```
Comment (denysbutenko) [inline src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html:68]: `ICU message format` means a different approach - https://github.com/truenas/webui#translating-text-to-other-languages

File: src/app/helptext/data-protection/cloudsync/cloudsync-form.ts [hunk 2]
```diff
@@ -178,4 +178,6 @@ export default {
 
   cloudsync_task_add: T('Add Cloud Sync Task'),
   cloudsync_task_edit: T('Edit Cloud Sync Task'),
+
+  empty_src_dirs_placeholder: T('Create empty source dirs on destination after sync'),
```
Comment (denysbutenko) [inline src/app/helptext/data-protection/cloudsync/cloudsync-form.ts:182]: Seems like unrelated to this PR

File: src/app/services/navigation/navigation.service.ts [hunk 3]
```diff
@@ -43,7 +43,8 @@ export class NavigationService {
       name: T('Datasets (WIP)'),
       type: MenuItemType.Link,
       tooltip: T('Datasets'),
-      icon: 'dns',
+      icon: 'dataset',
```
Comment (denysbutenko) [inline src/app/services/navigation/navigation.service.ts:46]: I left a [comment](https://ixsystems.atlassian.net/browse/NAS-117317?focusedCommentId=152138) under the ticket. I'm okay with keeping it

File: src/app/modules/layout/components/navigation/navigation.component.html [hunk 4]
```diff
@@ -47,9 +47,12 @@
           class="sidenav-link"
         >
           <span class="menu-item-tooltip" [matTooltip]="item.tooltip" matTooltipPosition="right"></span>
-          <mat-icon>{{ item.icon }}</mat-icon>
+          <mat-icon *ngIf="item.isSVGIcon; else matIcon" svgIcon="{{ item.icon }}" class="svg-icon"></mat-icon>
```
Comment (denysbutenko) [inline src/app/modules/layout/components/navigation/navigation.component.html:50]: `[svgIcon]="item.icon"` preferably

**Review by denysbutenko — CHANGES_REQUESTED** [review]

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html:68]: Done

File: src/app/services/navigation/navigation.service.ts (hunk 3 above)
Comment (AlexKarpov98) [reply src/app/services/navigation/navigation.service.ts:46]: Implemented as an ix-icon tag

File: src/app/modules/layout/components/navigation/navigation.component.html (hunk 4 above)
Comment (AlexKarpov98) [reply src/app/modules/layout/components/navigation/navigation.component.html:50]: Implemented as an ix-icon tag

Comment (AlexKarpov98) [reply src/app/modules/layout/components/navigation/navigation.component.html]: Replaced a MatIcon to a IxIcon

File: src/app/helptext/data-protection/cloudsync/cloudsync-form.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/helptext/data-protection/cloudsync/cloudsync-form.ts]: That is the code of another PR that already merged.
At the work beginning on this ticket I have pulled from a branch `NAS-117153` that was already merged, so the commits under that were added into this branch.

File: src/app/interfaces/menu-item.interface.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export interface MenuItem {
   name?: string; // Used as display text for item and title for separator type
   state?: string;
   icon?: string;
+  isSVGIcon?: boolean;
```
Comment (denysbutenko) [inline src/app/interfaces/menu-item.interface.ts:15]: Unused

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html [hunk 6]
```diff
@@ -64,9 +64,11 @@ <h3 mat-card-title>
       <div class="details-item">
         <span class="label">{{ 'User Quotas' | translate }}:</span>
         <ng-container *ngIf="checkQuotas; else noneTemplate">
-          <span class="value" *ngIf="!isLoadingQuotas; else valuePlaceholder">
-            {{ 'Quotas set for {n} users' | translate: { n: userQuotas } }}
-          </span>
+          <div *ngIf="!isLoadingQuotas; else valuePlaceholder">
```
Comment (denysbutenko) [inline src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html:67]: Is it intended to have an extra wrapper here and below?

**Review by denysbutenko — APPROVED** [review]
otherwise looks good 👍🏻

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html (hunk 6 above)
Comment (AlexKarpov98) [reply src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html:67]: 👍
It should be removed.
Done

File: src/app/interfaces/menu-item.interface.ts (hunk 5 above)
Comment (AlexKarpov98) [reply src/app/interfaces/menu-item.interface.ts:15]: 👍
Removed it.
